src/react-native-animated-charts/src/charts/linear/ChartPath.tsx

```diff
@@ -0,0 +1,381 @@
+
+    const setOriginData = useWorkletCallback(
+      (path: PathData, index?: number) => {
+        if (!path.data.length) {
+          return;
+        }
+
+        if (typeof index === 'undefined') {
+          originalX.value = '';
+          originalY.value = '';
+          return;
+        }
+
+        originalX.value = path.data[index].x.toString();
+        originalY.value = path.data[index].y.toString();
+      },
+      []
+    );
+
+    const resetGestureState = useWorkletCallback(() => {
+      originalX.value = '';
+      originalY.value = '';
+      isActive.value = false;
+      pathOpacity.value = withTiming(
+        1,
+        timingFeedbackConfig || timingFeedbackDefaultConfig
+      );
+      translationX.value = null;
+      translationY.value = null;
+    }, []);
+
+    useEffect(() => {
+      runOnUI(() => {
+        'worklet';
+        if (currentPath) {
+          setOriginData(currentPath);
+        }
+
+        if (currentPath?.path === previousPath?.path) {
+          return;
+        }
+
+        if (progress.value !== 0 && progress.value !== 1) {
+          cancelAnimation(progress);
+        }
+
+        progress.value = 0;
+
+        // this stores an instance of d3-interpolate-path on worklet side
+        // it means that we don't cross threads with that function
+        // which makes it super fast
+        if (previousPath && currentPath) {
+          const d3Interpolate = requireOnWorklet('d3-interpolate-path');
+
+          interpolatorWorklet().value = d3Interpolate.interpolatePath(
+            previousPath.path,
+            currentPath.path
+          );
+
+          progress.value = withDelay(
+            Platform.OS === 'ios' ? 0 : 100,
+            withTiming(1, timingAnimationConfig || timingAnimationDefaultConfig)
+          );
+        } else {
+          interpolatorWorklet().value = undefined;
+          progress.value = 1;
+        }
+      })();
+      // you don't need to change timingAnimationConfig that often
+      // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [
+      currentPath,
+      previousPath,
+      interpolatorWorklet,
+      progress,
+      setOriginData,
+    ]);
+
+    useAnimatedReaction(
+      () => ({ x: translationX.value, y: translationY.value }),
+      values => {
+        if (
+          !currentPath ||
+          !currentPath.parsed ||
+          progress.value === 0 ||
+          values.x === null ||
+          values.y === null
+        ) {
+          return;
+        }
+
+        const yForX = redash.getYForX(currentPath.parsed, Math.floor(values.x));
+
+        if (yForX !== null) {
+          positionY.value = yForX;
+        }
+
+        positionX.value = values.x;
+
+        const index = least(currentPath.points.length, i => {
+          if (typeof i === 'undefined' || values.x === null) {
+            return 0;
+          }
+
+          return Math.hypot(currentPath.points[i].x - Math.floor(values.x));
```

Comment:
why hypot? Can you add some comment?

Reply:
I used this code as an example.
https://observablehq.com/@d3/multi-line-chart
I can't say you right now why or is there any better way of doing it, but it works :)

Reply:
I don't find those charts working correctly (the one you linked). I think this is really worth understanding because we want to have the responsibility for those charts in the future when we will want to use them

`Math.hypot` function returns the square root of the sum of squares of its arguments. In this case, we have one arg so $\sqrt{x^2} = \abs{x}$.
Why can't we just use `Math.abs`?
 
Also, `Math.floor(values.x)` is def smaller or eq to `currentPath.points[i].x`, so why won't we just remove the `abs` then?

 ```suggestion
          return currentPath.points[i].x - Math.floor(values.x)
```

If this will not work, I will be super worried.

Reply:
As far as I understand, this will make more sense in R^3 space, but here I don't see what does this bring

Reply:
Okay, I am confused again `values.x` are values over x-axis? So `Math.floor(values.x)` isn't always `0`? If not, what are those `values.x`?